cmd/podman/images/scp.go

```diff
@@ -88,78 +111,119 @@ func scp(cmd *cobra.Command, args []string) (finalErr error) {
 	if err != nil {
 		return err
 	}
-	serv, err := parseArgs(args, cfg) // parses connection data and "which way" we are loading and saving
+	sourceInfo, destInfo, cliConnections, err := util.ParseImageSCPArgs(args) // parses connection data and "which way" we are loading and saving
 	if err != nil {
 		return err
 	}
+
+	if sourceInfo != nil { // we cannot import entities into util, map the options
+		sourceMarshal, err := json.Marshal(sourceInfo)
+		if err != nil {
+			return err
+		}
+		err = json.Unmarshal(sourceMarshal, &source)
```

Comment:
What does this code do? Is sourceInfo == source in the end?

Reply:
Nit if err = json ...

Reply:
so, entities cannot be imported into utils so I had to recreate the `ImageScpOptions` struct to pass the options properly. I think the same thing is done for saveOptions within utils

Reply:
for this one, i'd like to keep the function in utils but I think I can move the `Transfer` functions below out of utils.